### PR TITLE
try to use docker for test

### LINKED ISSUE #210: Accelerate unit_test_with_bcc
**Is your feature request related to a problem? Please describe.**
Accelerate [unit_test_with_bcc](https://github.com/sustainable-computing-io/kepler/blob/main/.github/workflows/unit_test.yml#L64-L83) 

**Describe the solution you'd like**
Currently the test spends much time on preparing the bcc on ubuntu. 

**Describe alternatives you've considered**
Since we already build bcc in [kepler base](https://github.com/sustainable-computing-io/kepler/blob/main/build/Dockerfile.base) container image, this could be optimized by running tests inside supported containers 

**Additional context**
@SamYuan1990 @cooktheryan

### PR DESCRIPTION
**Pull request by SamYuan1990**
resolve #210 

Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -58,34 +58,7 @@ jobs:
...
-    - name: Get ginkgo
-      run: make ginkgo-set
-      env:
-          GOPATH: /home/runner/go
-          GOBIN: /home/runner/go/bin
-    - name: Run 
-      run: |
-        make test-verbose
+    - name: build base image
+      run: docker build -f ./build/Dockerfile.base -t quay.io/sustainable_computing_io/kepler_base:latest-amd64 .
```
Comment (rootfs) [inline .github/workflows/unit_test.yml:62]: container image build could take time. If the dockerfile is not changed, we can just use the container image.

**Comment by SamYuan1990** [comment]
```
docker: Error response from daemon: invalid volume specification: '/home/runner/work/kepler/kepler:/': invalid mount config for type "bind": invalid specification: destination can't be '/'.
```
@rootfs , do you have any idea to run `make test-verbose` as a part of docker run command?
`cd /tmp && make test-verbose` works or not?

File: .github/workflows/unit_test.yml (hunk 1 above)
Comment (SamYuan1990) [reply .github/workflows/unit_test.yml]: ok

**Comment by rootfs** [comment]
> ```
> docker: Error response from daemon: invalid volume specification: '/home/runner/work/kepler/kepler:/': invalid mount config for type "bind": invalid specification: destination can't be '/'.
> ```
> 
> @rootfs , do you have any idea to run `make test-verbose` as a part of docker run command? `cd /tmp && make test-verbose` works or not?

@SamYuan1990 see if this helps (i am using podman and kepler_builder image since we need `make` to run the tests)

```bash
podman run -v $(pwd):/home quay.io/sustainable_computing_io/kepler_builder:go1.18 bash -c "cd /home; export PATH=/usr/local/go/bin:$PATH; make test-verbose"
```

**Review by rootfs — APPROVED** [review]

**Pull request merged**